src/vos/vos_obj.c

```diff
@@ -1992,22 +1991,11 @@ vos_obj_iter_aggregate(daos_handle_t ih, bool range_discard)
 		delete = true;
 		D_DEBUG(DB_IO, "Removing %s from tree\n",
 			iter->it_type == VOS_ITER_DKEY ? "dkey" : "akey");
-		/** Orphaned values indicate an incarnation log bug.  It happens
-		 *  when the key containing the subtree doesn't have a creation
-		 *  timestamp for updates in the subtree.
+
+		/* XXX: The value tree may be not empty because related prepared transaction can
```

Comment:
Remind me of the scenario here?   I'm not super keen on removing these checks if there is an alternative where we at least check the subtree for removed entries.    These checks have caught real bugs in the past.  Probably fine for a temporary solution.

Reply:
The patch will discard old DTX entries that will cause the ilog to be empty, but we do not remove related data record, instead, related data record is only marked as 'aborted', then it triggers assert dbtree_is_empty_inplace().